react/src/lib/components/DeckGLMap/utils/continuousLegend.ts

```diff
@@ -1,5 +1,86 @@
-import { scaleSequential, interpolateViridis, ScaleSequential } from "d3";
+import { color } from "d3-color";
+import { interpolateRgb } from "d3-interpolate";
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const colorTemplate = require("../../../../demo/example-data/welllayer_continuous_template.json");
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const colorTables = require("../../../../demo/example-data/color-tables.json");
 
-export function interpolatorContinuous(): ScaleSequential<string, never> {
-    return scaleSequential(interpolateViridis);
+interface colorTables {
+    name: string;
+    description: string;
+    colors: [number, number, number, number][];
+}
+
+interface propertiesObj {
+    objectName: string;
+    colorTable: string;
+    context: string;
+    colorInterpolation: string;
+}
+
+type propertiesArr = Array<propertiesObj>;
+
+interface colorTemplate {
+    name: string;
+    properties: propertiesArr;
+}
+
+export function colorsArray(
+    objectName: string
+): [number, number, number, number][] {
+    const properties = colorTemplate[0]["properties"];
+    const propertiesData = properties.filter(
+        (value: propertiesObj) => value.objectName == objectName
+    );
+    const colorTableData = colorTables.filter(
+        (value: colorTables) => value.name == propertiesData[0].colorTable
+    );
+    return colorTableData[0].colors;
+}
+
+export function rgbValues(
+    objectName: string,
+    point: number
+): number[] | { r: number; g: number; b: number; opacity: number } | undefined {
+    const color_table = colorsArray(objectName);
+    const colorArrays = color_table.find(
+        (value: [number, number, number, number]) => {
+            return point.toFixed(2) == value[0].toFixed(2);
```

Comment:
Comparing with 2 decimal points may not work for some log/property data.

Reply:
Removed toFixed(2) because as per the logic however, it checks if point and value matches else it will do interpolation.